From: Jiri Slaby <jslaby@suse.cz>
Date: Fri, 30 Aug 2024 12:25:21 +0200
Subject: perf trace: fix build on 32bit
Patch-mainline: submitted, <739001a4-4df1-4dec-a141-926c78c5c07e@kernel.org>
References: i586 + 6.11 build fix

i586 does not have HAVE_SYSCALL_TABLE_SUPPORT set. But we need
syscalltbl__id_at_idx(). Define it as identity.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
Fixes: 7a2fb5619cc1 (perf trace: Fix iteration of syscall ids in syscalltbl->entries)
Link: https://lore.kernel.org/all/6fe63fa3-6c63-4b75-ac09-884d26f6fb95@kernel.org/
---
 tools/perf/util/syscalltbl.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/tools/perf/util/syscalltbl.c b/tools/perf/util/syscalltbl.c
index 0dd26b99..75391ec4 100644
--- a/tools/perf/util/syscalltbl.c
+++ b/tools/perf/util/syscalltbl.c
@@ -178,6 +178,11 @@ int syscalltbl__id(struct syscalltbl *tbl, const char *name)
 	return audit_name_to_syscall(name, tbl->audit_machine);
 }
 
+int syscalltbl__id_at_idx(struct syscalltbl *, int idx)
+{
+	return idx;
+}
+
 int syscalltbl__strglobmatch_next(struct syscalltbl *tbl __maybe_unused,
 				  const char *syscall_glob __maybe_unused, int *idx __maybe_unused)
 {
-- 
2.46.0

